pkgs/applications/networking/msmtp/default.nix

```diff
@@ -1,59 +1,131 @@
-{ stdenv, lib, fetchurl, autoreconfHook, pkg-config, texinfo
-, netcat-gnu, gnutls, gsasl, libidn2, Security
-, withKeyring ? true, libsecret ? null
-, systemd ? null }:
+{ resholve
+, stdenv
+, symlinkJoin
+, lib
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, bash
+, coreutils
+, gnugrep
+, gnutls
+, gsasl
+, libidn2
+, netcat-gnu
+, texinfo
+, which
+, Security
+, withKeyring ? true
+, libsecret ? null
+, withSystemd ? stdenv.isLinux
+, systemd ? null
+}:
 
 let
-  tester = "n"; # {x| |p|P|n|s}
-  journal = if stdenv.isLinux then "y" else "n";
+  inherit (lib) getBin getExe optionals;
 
-in stdenv.mkDerivation rec {
-  pname = "msmtp";
   version = "1.8.20";
 
-  src = fetchurl {
-    url = "https://marlam.de/${pname}/releases/${pname}-${version}.tar.xz";
-    sha256 = "sha256-2TriqvwPSK99ydCzlN8buABYi4tOjQltizzyJTROsRE=";
+  src = fetchFromGitHub {
+    owner = "marlam";
+    repo = "msmtp-mirror";
+    rev = "msmtp-${version}";
+    hash = "sha256-RcQZ7Vm8UjJJoogkmUmZ+/2fz7C4AcVYY/kTOlfz7+I=";
```

Comment:
Personally, I would prefer it if we stuck to using the tarballs. I haven’t checked, but I would guess that downloading the tarball is going to use less bandwidth than even a shallow clone. More importantly, it would be more decentralized which I think is important. That being said, GitHub is probably more reliable, so this isn’t that big of a deal.

Reply:
`fetchFromGitHub` and friends are pretty efficient as they do fetch tarballs (or zips - don't recall) *and* with the upcoming changes to nix itself, it will support not having to actually unpack the archive but instead continue to work directly from the downloaded archive.

GH is less likely to be unavailable compared to $SOME_RANDOM_GUYS_OWN_SITE - that's really all.